src/setup_node_env/ensure_node_preserve_symlinks.js

```diff
@@ -99,6 +99,13 @@
     return 0;
   };
 
+  // Since we are using `stdio: inherit`, the child process will receive
+  // the `SIGINT` and `SIGTERM` from the terminal.
+  // However, we want the parent process not to exit until the child does.
+  // Adding the following handlers achieves that.
+  process.on('SIGINT', function () {});
+  process.on('SIGTERM', function () {});
```

Comment:
@mistic do you know how to test this appropriately? I've never seen any documentation that implies this is what we should do here, and I'm a little concerned about this.

Reply:
@spalger I think I've got why that is happening. The parent and the child processes are attached and signals are being inherited by the child too. On `ctrl+c` parent will exit but child will now only exit when the http-server graceful shutsdown. As long as this works on Windows as expected (which @afharo is testing) and as we are only using this on development I think we can move forward. Parent will always exit as soon as the child finishes as defined at https://github.com/elastic/kibana/pull/97223/files#diff-4fe511a7f9977dda595bb33284a2e238b03e2b673ff435bf718aaa805ff11b2fR110

Reply:
Tested on Windows. The same behaviour occurs:

- Running as `node scripts/kibana` will prevent the parent from stopping until the child has exited.
- Running as `yarn start` finishes the parent (detaching the process from the terminal), but the server's logs are still printed until fully finished.